src/main/webapp/components/InputFieldSelect.js

```diff
@@ -31,18 +101,18 @@ export const InputFieldSelect = hh(class InputFieldSelect extends Component {
         errorMessage: this.props.errorMessage,
         readOnly: this.props.readOnly,
         value: this.props.value,
-        currentValue: this.props.currentValue,
-        currentValueStr: this.props.currentValue.label,
+        currentValue: currentValue,
+        currentValueStr: currentValueStr,
         edited : edited
       }, [
           div({ className: "inputFieldSelectWrapper" }, [
             h(Select, {
               id: this.props.id,
               index: this.props.index,
               name: this.props.name,
-              value: this.props.value,
+              value: this.props.readOnly && (this.props.value === undefined || this.props.value === '') ? '--' : this.props.value,
               className: "inputFieldSelect",
-              onChange: this.props.onChange(this.props.index),
+              onChange: this.props.onChange(this.props.index, event),
```

Comment:
where this "event" came from ... It is undefined ... Review it and remove it if unused